github/git_commits.go

```diff
@@ -87,6 +90,29 @@ func (s *GitService) GetCommit(ctx context.Context, owner string, repo string, s
 	return c, resp, nil
 }
 
+// ListCommits fetches latest commits
+func (s *GitService) ListCommits(ctx context.Context, owner string, repo string, sha string) (*CommitList, *Response, error) {
```

Comment:
A pointer to a slice is generally to be avoided, as it is already a reference type.

Let's please change this return type to:

```
([]*Commit, *Response, error)
```

And delete lines 52-54 above.

Additionally, it is not idiomatic Go to say `owner string, repo string, sha string` in the parameter list.
Instead, please change this to `owner, repo, sha string`.